llvm/lib/Target/NVPTX/NVPTXISelLowering.cpp

```diff
@@ -2122,7 +2192,8 @@ SDValue NVPTXTargetLowering::LowerEXTRACT_VECTOR_ELT(SDValue Op,
   // Extract individual elements and select one of them.
   SDValue Vector = Op->getOperand(0);
   EVT VectorVT = Vector.getValueType();
-  assert(VectorVT == MVT::v2f16 && "Unexpected vector type.");
+  assert((VectorVT == MVT::v2f16 || VectorVT == MVT::v2i16) &&
```

Comment:
We may be missing a v2bf16 here.

Reply:
Moved to using Isv2x16Type so that addresses it.